pkg/cmd/server/api/types.go

```diff
@@ -144,6 +144,24 @@ type ProjectConfig struct {
 	// It is in the format namespace/template and it is optional.
 	// If it is not specified, a default template is used.
 	ProjectRequestTemplate string
+
+	// SecurityAllocator controls the automatic allocation of UIDs and MCS labels to a project. If nil, allocation is disabled.
+	SecurityAllocator *SecurityAllocator
+}
+
+type SecurityAllocator struct {
+	// UIDAllocatorRange defines the total set of Unix user IDs (UIDs) that will be allocated to projects automatically, and the size of the
+	// block each namespace gets. For example, 1000-1999/10 will allocate ten UIDs per namespace, and will be able to allocate up to 100 blocks
+	// before running out of space. The default is to allocate from 1 billion to 2 billion in 10k blocks (which is the expected size of the
+	// ranges Docker images will use once user namespaces are started).
+	UIDAllocatorRange string
```

Comment:
> On Jun 8, 2015, at 8:06 AM, David Eads <notifications@github.com> wrote:
> 
> In pkg/cmd/server/api/types.go:
> 
> > @@ -144,6 +144,24 @@ type ProjectConfig struct {
> >  	// It is in the format namespace/template and it is optional.
> >  	// If it is not specified, a default template is used.
> >  	ProjectRequestTemplate string
> > +
> > +	// SecurityAllocator controls the automatic allocation of UIDs and MCS labels to a project. If nil, allocation is disabled.
> > +	SecurityAllocator *SecurityAllocator
> > +}
> > +
> > +type SecurityAllocator struct {
> > +	// UIDAllocatorRange defines the total set of Unix user IDs (UIDs) that will be allocated to projects automatically, and the size of the
> > +	// block each namespace gets. For example, 1000-1999/10 will allocate ten UIDs per namespace, and will be able to allocate up to 100 blocks
> > +	// before running out of space. The default is to allocate from 1 billion to 2 billion in 10k blocks (which is the expected size of the
> > +	// ranges Docker images will use once user namespaces are started).
> > +	UIDAllocatorRange string
> Does it really make sense to include the block size in the same field? Not that I don't understand the format, but I haven't seen it commonly used before. It's very different from a CIDR, but it looks similar.
> 
So then we have two fields to persist and store everywhere.  It's really a cidr for uids, has the same pattern, but not distinct.
> —
> Reply to this email directly or view it on GitHub.
>